File: wordNet/src/main/java/org/sqlunet/wordnet/sql/SynsetQuery.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.wordnet.sql;

import android.database.sqlite.SQLiteDatabase;

import org.sqlunet.sql.DBQuery;

/**
 * Query for synsets containing a given word
 *
 * @author Bernard
 */
class SynsetQuery extends DBQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.SynsetQuery; // ;

	/**
	 * Constructor
	 *
	 * @param connection connection
	 * @param synsetId   target synset id
	 */
	@SuppressWarnings("boxing")
	public SynsetQuery(final SQLiteDatabase connection, final long synsetId)
	{
		super(connection, SynsetQuery.QUERY);
		setParams(synsetId);
	}

	/**
	 * Get synset id
	 *
	 * @return synset id
	 */
	public long getSynsetId()
	{
		assert this.cursor != null;
		return this.cursor.getLong(0);
	}

	/**
	 * Get synset definition
	 *
	 * @return definition
	 */
	public String getDefinition()
	{
		assert this.cursor != null;
		return this.cursor.getString(1);
	}

	/**
	 * Get synset domain id
	 *
	 * @return synset domain id
	 */
	public int getDomainId()
	{
		assert this.cursor != null;
		return this.cursor.getInt(2);
	}

	/**
	 * Get sample data
	 *
	 * @return samples as a semicolon-separated string
	 */
	public String getSample()
	{
		assert this.cursor != null;
		return this.cursor.getString(3);
	}
}

File: wordNet/src/main/java/org/sqlunet/wordnet/sql/SynsetQuery.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.wordnet.sql

import android.database.sqlite.SQLiteDatabase
import org.sqlunet.sql.DBQuery

/**
 * Query for synsets containing a given word
 *
 * @param connection connection
 * @param synsetId   target synset id
 *
 * @author Bernard
 */
class SynsetQuery(connection: SQLiteDatabase, synsetId: Long) : DBQuery(connection, QUERY) {

    init {
        setParams(synsetId)
    }

    /**
     * Synset id
     */
    val synsetId: Long
        get() = cursor!!.getLong(0)

    /**
     * Synset definition
     */
    val definition: String
        get() = cursor!!.getString(1)

    /**
     * Synset domain id
     */
    val domainId: Int
        get() = cursor!!.getInt(2)

    /**
     * Samples as a semicolon-separated string
     */
    val sample: String
        get() = cursor!!.getString(3)

    companion object {

        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.SynsetQuery
    }
}